OOPConsoleApp/OOP/OOP.cs

```diff
@@ -0,0 +1,266 @@
+                    $"{person.GetAge()}");
+            }
+
+            void Exercise2()
+            {
+                Console.WriteLine("Exercise 2.");
+
+                var student = new Student("Koval", "Irina", 19);
+                student.SetStudentAbility("musical ear");
+                student.SetStudentAssesment(5);
+
+                Console.WriteLine($"Student - {student.Surname}" +
+                    $" {student.Name}," +
+                    $" {student.Age} years old.");
+                Console.WriteLine($"Ability: {student.StudentAbility}. Assesment: {student.StudentAssesment}.");
+
+                var teacher = new Teacher("Ivanov", "Ivan", 41);
+                teacher.SetTeacherTemperament("sanguine");
+                teacher.SetTeacherSubject("Mathematicks");
+
+                Console.WriteLine($"Teacher - {teacher.PersonSurname} " +
+                    $"{teacher.PersonName}, " +
+                    $"{teacher.PersonAge} years old. ");
+                Console.WriteLine($"Temperament: {teacher.TeacherTemperament}. Subject: {teacher.TeacherSubject}.");
+            }
+            void Exercise3()
+            {
+                Console.WriteLine("Exercise 3.");
+
+                var personSecond = new Person("Datchin", "Valery", 43);
+
+                Console.WriteLine($"{personSecond.GetSurname()} {personSecond.GetName()}," +
+                $" {personSecond.GetAge()} y.o.");
+            }
+            void Exercise4()
+            {
+                Console.WriteLine("Exercise 4.");
+
+                var myPerson = new Person("Fleming", "Lise", 33);
+                myPerson.Print();
+
+                var myStudent = new Student("Koval", "Irina", 19);
+                myStudent.Print();
+
+                var myTeacher = new Teacher("Ivanov", "Ivan", 41);
+                myTeacher.Print();
+            }
+
+            void Exercise5()
+            {
+                Console.WriteLine("Exercise 5.");
+
+                var classroom = new Classroom();
+                classroom.AddPersons();
+                classroom.Print();
+                classroom.RemovePersons();
+                classroom.ClearPersons();
+            }
+        }
+    }
+}
+
+namespace OOP
+{
+    public class Classroom : IPrintable
+    {
+        private List<Person> _persons = new();
+
+        public void AddPersons()
+        {
+            var person1 = new Person("Pupkin", "Vasiliy", 20);
+            var person2 = new Person("Kirkorov", "Petya", 46);
+
+            _persons.Add(person1);
+            _persons.Add(person2);
+        }
+
+        public void RemovePersons()
+        {
+            _persons.RemoveAll(x => x == _persons[0]);
+        }
+
+        public void ClearPersons()
+        {
+            _persons.Clear();
+        }
+
+        public void Print()
+        {
+            foreach (var person in _persons)
+            {
+                string output = string.Join(" ", person.ToString());
```

Comment:
string.Join is a method that allows you to conveniently join collection elements into strings.
In this case, your collection is a string (since a string is an array of characters) and you glue characters together separated by spaces. There's a bug here.